src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -152,6 +155,11 @@ public Solution CurrentSolution
         /// </remarks>
         protected Solution SetCurrentSolution(Solution solution)
         {
+            if (solution is null)
```

Comment:
Legitimate bug found by this exercise. It would always have thrown because the solution.WithNewWorkspace would have triggered a null ref.